wagtail/snippets/views/snippets.py

```diff
@@ -779,210 +779,177 @@ def revisions_revert_view_class(self):
     def permission_policy(self):
         return ModelPermissionPolicy(self.model)
 
+    def get_common_view_kwargs(self, **kwargs):
+        return super().get_common_view_kwargs(
+            **{
+                "index_url_name": self.get_url_name("list"),
+                "index_results_url_name": self.get_url_name("list_results"),
+                "usage_url_name": self.get_url_name("usage"),
+                "history_url_name": self.get_url_name("history"),
+                "lock_url_name": self.get_url_name("lock"),
+                "unlock_url_name": self.get_url_name("unlock"),
+                "revisions_view_url_name": self.get_url_name("revisions_view"),
+                "revisions_revert_url_name": self.get_url_name("revisions_revert"),
+                "revisions_compare_url_name": self.get_url_name("revisions_compare"),
+                "revisions_unschedule_url_name": self.get_url_name(
+                    "revisions_unschedule"
+                ),
+                "unpublish_url_name": self.get_url_name("unpublish"),
+                **kwargs,
+            }
```

Comment:
Curious: is using `**{..., **kwargs}` (instead of passing them as kwargs directly) necessary to allow overriding an existing kwarg from a subclass?

Reply:
Yep, that's right - I had to introduce this pattern in `ModelViewSet.get_common_view_kwargs` so that `"index_url_name": self.get_url_name("index")` could be overridden with `"index_url_name": self.get_url_name("list")` here. Otherwise, it complains that `index_url_name` is duplicated in the kwargs list.